test/functional/ui/messages_spec.lua

```diff
@@ -1656,3 +1656,51 @@ aliquip ex ea commodo consequat.]])
     ]]}
   end)
 end)
+
+describe('ui/ext_messages on small windows', function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(20, 2)
+    screen:attach({rgb=true, ext_linegrid=true, ext_messages=true})
+    screen:set_default_attr_ids({
+      [1] = {bold = true, foreground = Screen.colors.Blue1},
+      [2] = {foreground = Screen.colors.Grey100, background = Screen.colors.Red},
+      [3] = {bold = true},
+      [4] = {bold = true, foreground = Screen.colors.SeaGreen4},
+      [5] = {foreground = Screen.colors.Blue1},
+      [6] = {bold = true, reverse = true},
+      [7] = {background = Screen.colors.Yellow},
+      [8] = {foreground = Screen.colors.Red},
+    })
+  end)
+  after_each(function()
+    os.remove('Xtest')
+  end)
+
+  it('ext_messages doesnt trigger unnecessary grid_scrolls', function()
```

Comment:
Is this really connected to ext_messages at all? don't we see the same change of behaviour without ext_messages but with three grid lines (so still two for the window) ?

Reply:
I had tried to reproduce it with the TUI and failed, but I just tried again and succeeded, so there's indeed nothing ext_messages-specific here.

In fact, Vim seems to have this bug too, so would you prefer that I fix it upstream and then port the patch to Neovim?

Reply:
It would make sense to report the issue upstream yes, but if the fix is the same you can just add the vim-patch tag to the commit message.